fmpz_mod_poly/set_nmod_poly.c

```diff
@@ -0,0 +1,34 @@
+/*
+    Copyright (C) 2011 Sebastian Pancratz
+
+    This file is part of FLINT.
+
+    FLINT is free software: you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 2.1 of the License, or
+    (at your option) any later version.  See <https://www.gnu.org/licenses/>.
+*/
+
+#include <gmp.h>
+#include "flint.h"
+#include "fmpz.h"
+#include "fmpz_mod_poly.h"
+
+void fmpz_mod_poly_set_nmod_poly(fmpz_mod_poly_t f, const nmod_poly_t g)
+{
+    slong i;
+    
+    /* fit_length */
+    if (f->alloc < g->length)
```

Comment:
Why not fmpz_mod_poly_fit_length?

Reply:
That requires a context object, which propagates the requirement right up through all the other functions, meaning we end up with functions with two context objects that have to be passed. I prefer to avoid using it, as it is not needed.

Reply:
Yup. This is going to be a function that requires manual attention if the internals of fmpz_mod_poly ever change.